### PR TITLE
Make Microsoft.CodeAnalysis.* nugets PrivateAssets

### PR DESCRIPTION
**Pull request by LucGenetier**

### CODE REVIEW

**Comment by LucGenetier** [comment]
✅ No public API change.

**Review by CarlosFigueiraMSFT — APPROVED** [review]
:shipit:

**Review by MikeStall — APPROVED** [review]

**Comment by LucGenetier** [comment]
✅ No public API change.

File: src/Directory.Build.props [hunk 1]
```diff
@@ -24,8 +24,8 @@
 
   <ItemGroup>
     <PackageReference Include="StyleCop.Analyzers" Version="1.1.118" PrivateAssets="all" />
-    <PackageReference Include="Microsoft.CodeAnalysis" Version="4.2.0" />
-    <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="7.0.0">
+    <PackageReference Include="Microsoft.CodeAnalysis" Version="4.2.0" PrivateAssets="all" />
+    <PackageReference Include="Microsoft.CodeAnalysis.NetAnalyzers" Version="7.0.0" >
       <PrivateAssets>all</PrivateAssets>
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
```
Comment (MikeStall) [inline src/Directory.Build.props:30]: What does this line mean? do we need to do that too?

Comment (LucGenetier) [reply src/Directory.Build.props:30]: It's automatically added like this when you add the package AFAIR

**Pull request merged**